azurelinuxagent/ga/exthandlers.py

```diff
@@ -652,6 +674,8 @@ def __init__(self, ext_handler, protocol):
         self.is_upgrade = False
         self.logger = None
         self.set_logger()
+        self.handler_env = {EXTENSION_PATH: self.get_base_dir(),
```

Comment:
why not explicitly define those variables in each of the commands (update, enable, etc)? By adding this member now we need to be sure its state is kept update as the workflow progresses, plus is a little hard to follow which variables are set on which operations

Reply:
A new object of ExtHandlerInstance is created everytime the agent starts extension processing so the state maintenance is only for that particular goalstate call, which is easier to maintain